### ISSUE TITLE
Feed time-distributed Dense layer into an (LSTM) RNN

### ISSUE THREAD

**Issue by kdx2** [keras-team/keras#12114]
Dear everyone,
I am trying to make an LRCN model and as my results from training are very bad, e.g.:

Adam
> Epoch 1/5
> 10390/10390 [==============================] - 276s 27ms/step - loss: 0.6402 - acc: 0.7147 - val_loss: 1.3077 - val_acc: 0.2504
> Epoch 2/5
> 10390/10390 [==============================] - 265s 25ms/step - loss: 0.2148 - acc: 0.9116 - val_loss: 1.7828 - val_acc: 0.3985
> Epoch 3/5
> 10390/10390 [==============================] - 268s 26ms/step - loss: 0.1428 - acc: 0.9380 - val_loss: 1.7528 - val_acc: 0.4105
> Epoch 4/5
> 10390/10390 [==============================] - 271s 26ms/step - loss: 0.1078 - acc: 0.9503 - val_loss: 2.5492 - val_acc: 0.3847
> Epoch 5/5
> 10390/10390 [==============================] - 267s 26ms/step - loss: 0.0877 - acc: 0.9589 - val_loss: 3.0907 - val_acc: 0.3623

RMSprop
> Epoch 1/5
> 10390/10390 [==============================] - 1742s 168ms/step - loss: 1.1163 - acc: 0.3443 - val_loss: 1.0252 - val_acc: 0.6360
> Epoch 2/5
> 10390/10390 [==============================] - 1684s 162ms/step - loss: 1.1118 - acc: 0.3556 - val_loss: 1.1023 - val_acc: 0.1282
> Epoch 3/5
> 10390/10390 [==============================] - 1728s 166ms/step - loss: 1.1126 - acc: 0.3556 - val_loss: 0.9226 - val_acc: 0.6360
> Epoch 4/5
> 10390/10390 [==============================] - 1737s 167ms/step - loss: 1.1131 - acc: 0.3502 - val_loss: 0.9797 - val_acc: 0.6360
> Epoch 5/5
> 10390/10390 [==============================] - 1824s 176ms/step - loss: 1.1109 - acc: 0.3497 - val_loss: 1.2216 - val_acc: 0.2358

*over 100 epochs, nothing changes... even the val_acc numbers are the same. Fluctuating... random guessing.

I am just wondering if TimeDistributed is doing what I think is doing. Namely, if it is connecting to the RNN properly.

I pass a 5D tensor, of the shape (1, numFrames, height, width, channels) to the NN.
All convolutional and max-pooling layers are TimeDistributed.
In the end I have:

> model.add(TimeDistributed(Dense(64, activation="relu")))
> model.add(LSTM(units=256, return_sequences=False))

Is TimeDistributed here making a tensor such that each of the TimeDistributed operations before it extend a dimension in that tensor such that then, each of them is sliced and in the same sequence fed into the RNN input nodes? Hence, very roughly speaking, is TimeDistributed here making an individual CNN per frame and then feeding its Dense layer derivation, sequentially into the inputs of the RNN?

If that is true, could you please hint me what else I can do to make that architecture start learning? My dataset is not open to the open public and is very small. Around 92 000 frames, roughly 100 videos of only 16 people and is imbalanced. I added Dropout, aggressive Dropout (e.g.0.8-0.9), Batch norm, tried 3 optimizers but nothing seems to help.
That is why I am wondering maybe TimeDistributed is not doing what I imagine it's doing.
If you could give me an idea, expertise, I would be happy to find it out.
